nft-block.js

```diff
@@ -78,8 +74,39 @@ module.exports = class NftBlock extends Block {
 
     // Adding NFT to artists list.
     let ownedNfts = this.nftOwnerMap.get(owner) || [];
-    ownedNfts.push(nftID);
-    this.nftOwnerMap.set(owner, ownedNfts);
+    if(ownedNfts.includes(nftID) == false)
+    {
+      ownedNfts.push(nftID);
+      this.nftOwnerMap.set(owner, ownedNfts);
+    }
+  }
+
+  transferNft(owner, nftID, sender)
+  {
+    let ownedNftsO = this.nftOwnerMap.get(owner) || [];
```

Comment:
Perhaps add a comment or rename variables to indicate who is sending and receiving the NFT.